processes/processes.go

```diff
@@ -186,165 +273,169 @@ func (procPlg *procPlugin) CollectMetrics(metricTypes []plugin.MetricType) ([]pl
-						Unit_:        metricNames[metricName].unit,
-						Description_: metricNames[metricName].description,
+			// return aggregated metrics
+			if reqProcPID == "all" {
+				for aggrProcessName, aggrMetric := range aggregated {
+					for aggrMetricName, aggrData := range aggrMetric {
+						if reqProcName == aggrProcessName || reqProcName == "*" {
+							if metricName == aggrMetricName {
+								nuns := append([]plugin.NamespaceElement{}, ns...)
+								nuns[4] = fillNsElement(&nuns[4], aggrProcessName)
+								nuns[5] = fillNsElement(&nuns[5], "all")
+								metrics = append(metrics, prepareMetric(nuns, metricName, aggrData))
+							}
+						}
 					}
-					metrics = append(metrics, metric)
-				} else {
-					return nil, serror.New(fmt.Errorf("Stat name {%s} not found!", metricName))
 				}
 			}
+		} else if len(ns) == 6 && ns[3].Value == "process" { // process count
+			reqProcName := ns[4].Value
+			metricName := ns[5].Value
 
-		} else if len(ns) == 4 && str.Contains(States.Values(), ns[3].Value) {
-			state := ns[3].Value
-			if val, ok := stateCount[state]; ok {
-				metric := plugin.MetricType{
-					Namespace_:   ns,
-					Data_:        val,
-					Timestamp_:   time.Now(),
-					Description_: fmt.Sprintf("Number of processes in %s state", state),
-					Unit_:        "",
+			for processName, process := range stats {
+				processCount[processName] = len(process)
+			}
+
+			// return process count metric
+			for processName, processCount := range processCount {
+				if reqProcName == processName || reqProcName == "*" {
+					nuns := append([]plugin.NamespaceElement{}, ns...)
+					nuns[4] = fillNsElement(&nuns[4], processName)
+					nuns[5] = fillNsElement(&nuns[5], "ps_count")
+					metrics = append(metrics, prepareMetric(nuns, metricName, processCount))
 				}
-				metrics = append(metrics, metric)
-			} else {
-				return nil, serror.New(fmt.Errorf("Unknown namespace."))
 			}
+		} else if len(ns) == 5 && ns[3].Value == "state" { // globally aggregated process states
+			metricName := ns[4].Value
 
+			for stateName, val := range stateCount {
+				if metricName == stateName {
+					nuns := append([]plugin.NamespaceElement{}, ns...)
+					metrics = append(metrics, prepareMetric(nuns, metricName, val))
+				}
+			}
 		} else {
-			return nil, serror.New(fmt.Errorf("Unknown namespace."))
+			return nil, fmt.Errorf("Bad namespace: %s", strings.Join(ns.Strings(), "/"))
 		}
 	}
 
 	return metrics, nil
 }
 
-func setProcMetrics(instances []Proc) map[string]uint64 {
-	procMetrics := map[string]uint64{}
-	for metricName, _ := range metricNames {
-		procMetrics[metricName] = 0
-	}
-	procMetrics["ps_count"] = uint64(len(instances))
+func setProcMetrics(instance Proc) map[string]interface{} {
+	var procMetrics = make(map[string]interface{})
 
-	for _, instance := range instances {
-		vm, _ := strconv.ParseUint(string(instance.Stat[22]), 10, 64)
-		procMetrics["ps_vm"] += vm
+	vm, _ := strconv.ParseUint(string(instance.Stat[22]), 10, 64)
```

Comment:
Why errors returned by `strconv` are not handled? I suggest handling of this errors.